### PR TITLE
Adding fsdp gpt2 training scripts

### PR DESCRIPTION
**Pull request by amithrm**

### CODE REVIEW

File: torch-neuronx/training/fsdp_gpt2/run_clm_slurm.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+
+export TRAINING_PRECISION=$1 #options FP32, BF16, MIXED
+
+if [[ "BF16" == $TRAINING_PRECISION ]]; then
+    echo "USING BF16 ONLY"
+    export XLA_USE_BF16=1
+    # need to clean up later, as 
+    # --distribution-strategy=FSDP
+    export NEURON_CC_FLAGS="--retry_failed_compilation --disable-internal-data-race-checker --internal-dram-page-size=2048 --enable-experimental-spmd --internal-allreduce-buffer-size=135 --model-type transformer --tensorizer-options=\'--enable-real-yk-ccop\' --internal-max-instruction-limit=10000000 --enable-experimental-O1  --internal-layers-per-module 4 --internal-ccop-bucketing --internal-ccop-bucketing-allgather-size-in-bytes 1 --internal-ccop-bucketing-reducescatter-size-in-bytes 62481600 --internal-ccop-bucketing-allreduce-size-in-bytes 62481600 --internal-build-with-users"
```
Comment (jyang-aws) [inline torch-neuronx/training/fsdp_gpt2/run_clm_slurm.sh:39]: The flags need to be cleaned up just like the `run_clm_neo.sh` file

File: torch-neuronx/training/fsdp_gpt2/setup_gpt2_model_dataset.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (jyang-aws) [inline torch-neuronx/training/fsdp_gpt2/setup_gpt2_model_dataset.sh:1]: This file is actually for our pipeline setup (to grab the cache from s3), may not need it for external customer

File: torch-neuronx/training/fsdp_gpt2/run_clm_slurm_mics.sh [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+wait %1
+
+ret_val=$?
+if [ $ret_val -eq 0 ]; then
+    success=1
+else
+    success=0
+fi
+
+dump_to_s3_update_json_scr=../../../dump_to_s3_update_test_json.sh
```
Comment (jyang-aws) [inline torch-neuronx/training/fsdp_gpt2/run_clm_slurm_mics.sh:120]: this is for pipeline, in this sample repo, I don't think we have `dump....json.sh`

File: torch-neuronx/training/fsdp_gpt2/run_clm_slurm.sh (hunk 1 above)
Comment (amithrm) [reply torch-neuronx/training/fsdp_gpt2/run_clm_slurm.sh:39]: Thanks Jie! working on it..

**Review by kct22aws — APPROVED** [review]

**Review by 5cp — APPROVED** [review]
LGTM

**Pull request merged**